### DISCUSSION TITLE
Add support for Libravatar

### DISCUSSION THREAD

**Discussion by xesjkeee** [directus/directus#5103] [Feature Requests]
Would be very useful when you have [Libravatar](https://www.libravatar.org/) server for responding users avatars

**Comment by rijkvanzanten** [comment]
We used to have Gravatar for this purpose, but that was removed after people started reporting that that's breaking GDPR as you're sending up the user's email address to an (unauthorized) third party service 🤔

**Reply by benhaynes** [reply]
Agree. While I like the idea of these global avatar services... I think I'd prefer to have a lightweight avatar generator in the platform (been thinking about this one lately) and/or an opt-in "fetch libravatar/gravatar" button for users.

**Reply by xesjkeee** [reply]
Now I have log in to Admin App via corporate SSO provider, thats mean all my users agree to use himself corporate email to fetch avatars from corporate libravatar server

What do you think how can I fetch avatars now for my organization users only?

**Reply by rijkvanzanten** [reply]
The upcoming `oauth` hook could be a good place to dynamically update the user that logs in with the libravatar avatar 🤔 https://github.com/directus/directus/pull/5045

**Reply by joeinnes** [reply]
@benhaynes in terms of exploring the avatar generator, I've used https://robohash.org successfully before, and they are open source (MIT), although it's Python so wouldn't be drop in... (I found this, not sure if it would be 'drop in' either, but it's supposedly a Node implementation, also MIT: https://github.com/csvwolf/robohash) but you can send any string to Robohash to convert to an avatar. GDPR violations could be avoided by sending either the user ID (or if you're really GDPaRanoid, a hash of the user ID).

For using an external avatar provider, you would necessarily need to share personal information to the third party in order for the third party to identify whether a user has an account or not, so I think this would definitely need to be opt-in as you suggest. The problem is that I suspect most users would never bother, unless there were some kind of onboarding wizard which prompts users on first log in to set up various bits of account information (eg: avatar, title, role, etc.).

**Reply by benhaynes** [reply]
Yeah, I think this could easily be included in the (future) User Onboarding Flow (ref: https://github.com/directus/directus/discussions/3080). Also, being a designer, I have some ideas for a Directus-specific avatar generator using interchangeable vector assets. I'd like to keep our avatars fun but professional, which is a hard balance to strike.

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
